# SeqAudit: Citations
If you use the SeqAudit pipeline, please cite the following tools and resources:
Core Pipeline Framework

### [nf-core](https://pubmed.ncbi.nlm.nih.gov/32055031/)

> Ewels PA, Peltzer A, Fillinger S, Patel H, Alneberg J, Wilm A, Garcia MU, Di Tommaso P, Nahnsen S. The nf-core framework for community-curated bioinformatics pipelines. Nat Biotechnol. 2020 Mar;38(3):276-278. doi: 10.1038/s41587-020-0439-x. PubMed PMID: 32055031.

### [Nextflow](https://pubmed.ncbi.nlm.nih.gov/28398311/)

> Di Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C. Nextflow enables reproducible computational workflows. Nat Biotechnol. 2017 Apr 11;35(4):316-319. doi: 10.1038/nbt.3820. PubMed PMID: 28398311.

## Sequencing Data Simulation Tools

### [PBSIM3](https://doi.org/10.1093) (ONT and PacBio simulation)

    Ono, Y., Asai, K., & Hamada, M. (2021). PBSIM2: a simulator for long-read sequencers with a novel generative model of quality scores. Bioinformatics, 37(5), 589-595. https://doi.org/10.1093/bioinformatics/btaa835
    Ono, Y., Hamada, M., & Asai, K. (2013). PBSIM: PacBio reads simulator--toward accurate genome assembly. Bioinformatics, 29(1), 119-121. https://doi.org/10.1093/bioinformatics/bts649

### [NanoSim](https://doi.org/10.1093/gigascience/gix010) (ONT simulation alternative)

    Yang, C., Chu, J., Warren, R. L., & Birol, I. (2017). NanoSim: nanopore sequence read simulator based on statistical characterization. GigaScience, 6(4), gix010. https://doi.org/10.1093/gigascience/gix010

### [ART](https://doi.org/10.1093/bioinformatics/btr708) (Illumina simulation)

    Huang, W., Li, L., Myers, J. R., & Marth, G. T. (2012). ART: a next-generation sequencing read simulator. Bioinformatics, 28(4), 593-594. https://doi.org/10.1093/bioinformatics/btr708


## Data Retrieval and Processing

### [NCBI Entrez Direct](https://www.ncbi.nlm.nih.gov/books/NBK179288/)

    Kans, J. (2013). Entrez Direct: E-utilities on the Unix Command Line. In Entrez Programming Utilities Help [Internet]. Bethesda (MD): National Center for Biotechnology Information (US). Available from: https://www.ncbi.nlm.nih.gov/books/NBK179288/

### [MultiQC](https://doi.org/10.1093/bioinformatics/btw354) (Quality Control Reporting)

    Ewels, P., Magnusson, M., Lundin, S., & Käller, M. (2016). MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics, 32(19), 3047-3048. https://doi.org/10.1093/bioinformatics/btw354

## Container and Environment Management

### Docker

    Merkel, D. (2014). Docker: lightweight linux containers for consistent development and deployment. Linux Journal, 2014(239), 2.

### [Singularity](https://doi.org/10.1371/journal.pone.0177459)

    Kurtzer, G. M., Sochat, V., & Bauer, M. W. (2017). Singularity: Scientific containers for mobility of compute. PLoS ONE, 12(5), e0177459. https://doi.org/10.1371/journal.pone.0177459

### [Conda](https://docs.anaconda.com/)

    Anaconda Software Distribution. (2020). Anaconda Documentation. Anaconda Inc. Retrieved from https://docs.anaconda.com/

### [Bioconda](https://doi.org/10.1038/s41592-018-0046-7)

    Grüning, B., Dale, R., Sjödin, A., Chapman, B. A., Rowe, J., Tomkins-Tinch, C. H., ... & Köster, J. (2018). Bioconda: sustainable and comprehensive software distribution for the life sciences. Nature Methods, 15(7), 475-476. https://doi.org/10.1038/s41592-018-0046-7

## Wave Container Provisioning

### [Wave](https://seqera.io/wave/)

    Seqera Labs. (2023). Wave: Container provisioning service for Nextflow pipelines. Retrieved from https://seqera.io/wave/

### Additional Resources
## Reference Databases

### [NCBI RefSeq](https://doi.org/10.1093/nar/gkv1189)

    O'Leary, N. A., Wright, M. W., Brister, J. R., Ciufo, S., Haddad, D., McVeigh, R., ... & Pruitt, K. D. (2016). Reference sequence (RefSeq) database at NCBI: current status, taxonomic expansion, and functional annotation. Nucleic Acids Research, 44(D1), D733-D745. https://doi.org/10.1093/nar/gkv1189

### [NCBI GenBank](https://doi.org/10.1093/nar/gkw1070)

    Benson, D. A., Cavanaugh, M., Clark, K., Karsch-Mizrachi, I., Lipman, D. J., Ostell, J., & Sayers, E. W. (2017). GenBank. Nucleic Acids Research, 45(D1), D37-D42. https://doi.org/10.1093/nar/gkw1070

Citation Format

When citing the SeqAudit pipeline in your publications, please use the following format:
We used the SeqAudit pipeline (version X.X.X) for simulating [ONT/PacBio/Illumina] sequencing data.
The pipeline utilizes [PBSIM3/NanoSim] for long-read simulation and ART for Illumina read simulation,
implemented in Nextflow (Di Tommaso et al., 2017).
Software Versions

The specific versions of software used in your analysis can be found in the MultiQC report generated by the pipeline, or in the individual versions.yml files produced by each process.
Acknowledgments

We thank the developers and maintainers of all the open-source tools that make the SeqAudit pipeline possible. Special thanks to the Nextflow community and the Bioconda project for providing standardized, reproducible software environments.

## Software packaging/containerisation tools

- [Anaconda](https://anaconda.com)

  > Anaconda Software Distribution. Computer software. Vers. 2-2.4.0. Anaconda, Nov. 2016. Web.

- [Bioconda](https://pubmed.ncbi.nlm.nih.gov/29967506/)

  > Grüning B, Dale R, Sjödin A, Chapman BA, Rowe J, Tomkins-Tinch CH, Valieris R, Köster J; Bioconda Team. Bioconda: sustainable and comprehensive software distribution for the life sciences. Nat Methods. 2018 Jul;15(7):475-476. doi: 10.1038/s41592-018-0046-7. PubMed PMID: 29967506.

- [BioContainers](https://pubmed.ncbi.nlm.nih.gov/28379341/)

  > da Veiga Leprevost F, Grüning B, Aflitos SA, Röst HL, Uszkoreit J, Barsnes H, Vaudel M, Moreno P, Gatto L, Weber J, Bai M, Jimenez RC, Sachsenberg T, Pfeuffer J, Alvarez RV, Griss J, Nesvizhskii AI, Perez-Riverol Y. BioContainers: an open-source and community-driven framework for software standardization. Bioinformatics. 2017 Aug 15;33(16):2580-2582. doi: 10.1093/bioinformatics/btx192. PubMed PMID: 28379341; PubMed Central PMCID: PMC5870671.

- [Docker](https://dl.acm.org/doi/10.5555/2600239.2600241)

  > Merkel, D. (2014). Docker: lightweight linux containers for consistent development and deployment. Linux Journal, 2014(239), 2. doi: 10.5555/2600239.2600241.

- [Singularity](https://pubmed.ncbi.nlm.nih.gov/28494014/)

  > Kurtzer GM, Sochat V, Bauer MW. Singularity: Scientific containers for mobility of compute. PLoS One. 2017 May 11;12(5):e0177459. doi: 10.1371/journal.pone.0177459. eCollection 2017. PubMed PMID: 28494014; PubMed Central PMCID: PMC5426675.
